File: BikeRental/BikeRentalClient/BikeRentalClient/BikeUtils/Bike.cs
using System.Text.Json.Serialization;

namespace BikeRentalClient.BikeUtils
{
    public class Bike
    {
        public int Bike_id { get; set; } = 0;
        public int Manufacturer_id { get; set; } = 0;
        public string Model { get; set; } = string.Empty;

        public int Year { get; set; } = 0;
        public float Price { get; set; } = 0.0f;

        [JsonConverter(typeof(JsonStringEnumConverter))]
        public BikeStatus Status {  get; set; }

        public int? CurrentRenter_id { get; set; } = null;
        public DateTime? RentDate { get; set; } = null;
        public int? RentTime { get; set; } = null;
        public enum BikeStatus
        {
            AVAILABLE,
            RENTED,
            MAINTENANCE
        }
    }
}

File: BikeRental/BikeRentalServer/src/main/java/com/example/BikeRental/BikeUtils/Bike.java
package com.example.BikeRental.BikeUtils;

import com.example.BikeRental.ManufacturerUtils.Manufacturer;
import com.example.BikeRental.UserUtils.User;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Entity
@Table(name="bikes")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Bike {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private Integer bike_id;

    @Column(nullable = false)
    private Integer manufacturer_id;

    @Column(nullable = false)
    private String model;

    @Column(nullable = false)
    private Integer year;

    @Column(nullable = false)
    private Float price;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private BikeStatus status;

    private Integer currentRenter_id;
    private Date rentDate;

    public enum BikeStatus{
        AVAILABLE,
        RENTED,
        MAINTENANCE
    }
}
